lib/queue.rb

```diff
@@ -22,10 +53,17 @@ def size
   end
 
   def empty?
-    raise NotImplementedError, "Not yet implemented"
+    @front == -1
   end
 
   def to_s
-    return @store.to_s
+    list = []
+    index = @front
+    while index <= @rear 
+      list.push(@queue[index])
+      index += 1
```

Comment:
Otherwise you won't be able to wrap around. 

```suggestion
      index = (index + 1) % @store.length
```